test/UnitWOPIHttpRedirect.cpp

```diff
@@ -0,0 +1,151 @@
+    {
+        Poco::URI uriReq(request.getURI());
+        Poco::RegularExpression regInfo("/wopi/files/1");
+        std::string redirectUri = "/wopi/files/0";
+        Poco::RegularExpression regRedirected(redirectUri);
+
+        Log::StreamLogger logger = Log::info();
+        if (logger.enabled())
+        {
+            logger << "Fake wopi host request URI [" << uriReq.toString() << "]:\n";
+            for (const auto& pair : request)
+            {
+                logger << '\t' << pair.first << ": " << pair.second << " / ";
+            }
+
+            LOG_END(logger, true);
+        }
+
+        // CheckFileInfo - returns redirect response
+        if (request.getMethod() == "GET" && regInfo.match(uriReq.getPath()))
+        {
+            LOG_INF("Fake wopi host request, handling CheckFileInfo (1/2): redirect");
+
+            assertCheckFileInfoRequest(request);
+
+            std::ostringstream oss;
+            oss << "HTTP/1.1 302 Found\r\n"
+                "Location: " << helpers::getTestServerURI() << redirectUri << "?" << params << "\r\n"
+                "\r\n";
+
+            socket->send(oss.str());
+            socket->shutdown();
+
+            return true;
+        }
+        // CheckFileInfo - for redirected URI
+        else if (request.getMethod() == "GET" && regRedirected.match(uriReq.getPath()))
+        {
+            LOG_INF("Fake wopi host request, handling CheckFileInfo: (2/2): success");
+
+            assertCheckFileInfoRequest(request);
+
+            Poco::LocalDateTime now;
+            const std::string fileName(uriReq.getPath() == "/wopi/files/3" ? "he%llo.txt" : "hello.txt");
+            Poco::JSON::Object::Ptr fileInfo = new Poco::JSON::Object();
+            fileInfo->set("BaseFileName", fileName);
+            fileInfo->set("Size", _fileContent.size());
+            fileInfo->set("Version", "1.0");
+            fileInfo->set("OwnerId", "test");
+            fileInfo->set("UserId", "test");
+            fileInfo->set("UserFriendlyName", "test");
+            fileInfo->set("UserCanWrite", "true");
+            fileInfo->set("PostMessageOrigin", "localhost");
+            fileInfo->set("LastModifiedTime", Util::getIso8601FracformatTime(_fileLastModifiedTime));
+            fileInfo->set("EnableOwnerTermination", "true");
+
+            std::ostringstream jsonStream;
+            fileInfo->stringify(jsonStream);
+            std::string responseString = jsonStream.str();
+
+            const std::string mimeType = "application/json; charset=utf-8";
+
+            std::ostringstream oss;
+            oss << "HTTP/1.1 200 OK\r\n"
+                "Last-Modified: " << Util::getHttpTime(_fileLastModifiedTime) << "\r\n"
+                "User-Agent: " WOPI_AGENT_STRING "\r\n"
+                "Content-Length: " << responseString.size() << "\r\n"
+                "Content-Type: " << mimeType << "\r\n"
+                "\r\n"
+                << responseString;
```

Comment:
We will convert these to http::Response at some point. Feel free to use it here (grep for samples).

Reply:
I don't see any 'http::Response', does it exist even in co-6-4-9? (before async save)